libwallet/assets/btc/sync.go

```diff
@@ -191,26 +191,30 @@ func (asset *Asset) updateSyncProgress(rawBlockHeight int32) {
 	asset.syncData.headersFetchProgress.HeadersFetchProgress = int32((headersFetchedSoFar * 100) / allHeadersToFetch)
 	asset.syncData.headersFetchProgress.GeneralSyncProgress.TotalSyncProgress = asset.syncData.headersFetchProgress.HeadersFetchProgress
 	asset.syncData.headersFetchProgress.GeneralSyncProgress.TotalTimeRemainingSeconds = int64((timeSpentSoFar * remainingHeaders) / headersFetchedSoFar)
+	asset.syncData.mu.Unlock()
 
 	// publish the sync progress results to all listeners.
+	asset.syncData.mu.RLock()
 	for _, listener := range asset.syncData.syncProgressListeners {
 		if listener.OnHeadersFetchProgress != nil {
 			listener.OnHeadersFetchProgress(&asset.syncData.headersFetchProgress)
 		}
 	}
+	asset.syncData.mu.RUnlock()
 }
 
 func (asset *Asset) publishHeadersFetchComplete() {
 	asset.syncData.mu.Lock()
-	defer asset.syncData.mu.Unlock()
-
-	asset.handleSyncUIUpdate()
-
 	asset.syncData.synced = true
 	asset.syncData.syncing = false
+	asset.syncData.mu.Unlock()
+
+	asset.handleSyncUIUpdate()
 }
 
 func (asset *Asset) handleSyncUIUpdate() {
+	asset.syncData.mu.RLock()
+	defer asset.syncData.mu.RUnlock()
 	for _, listener := range asset.syncData.syncProgressListeners {
```

Comment:
I don't think this was causing any problems but it follows the same problem pattern causing the freezes so changing just in case. If a read lock is required in OnSyncCompleted that is now OK.